rules_poetry/poetry.bzl

```diff
@@ -105,7 +105,14 @@ def _impl(repository_ctx):
 
     lockfile = json.decode(result.stdout)
     metadata = lockfile["metadata"]
-    if "files" in metadata:  # Poetry 1.x format
+    package_files = [{"name": package["name"], "files": package["files"]} for package in lockfile["package"] if package.get("files")]
```

Comment:
I would go for explicit version checking:
```
if metadata["lock-version"] in ["2.0"]:
```
etc. as far back as they had the `lock-version` item in `metadata`